src/Microsoft.PowerShell.Commands.Utility/commands/utility/FormatAndOutput/format-hex/Format-Hex.cs

```diff
@@ -17,12 +17,30 @@ namespace Microsoft.PowerShell.Commands
     /// Displays the hexadecimal equivalent of the input data.
```

Comment:
@vexx32, your last commit had 7 failures in `PowerShell-CI-linux`
(These are 5 of the failures)

FormatHex.Cmdlet Functionality.Validate that streamed text does not have buffer underrun problems ''a' * 30 | Format-Hex'
```powershell
Expected the value to have type [Microsoft.PowerShell.Commands.ByteCollection] or any of its subtypes, but got @(0000000000000000   61 61 61 61 61 61 61 61 61 61 61 61 61 61 61 61  aaaaaaaaaaaaaaaa, 0000000000000010   61 61 61 61 61 61 61 61 61 61 61 61 61 61        aaaaaaaaaaaaaa  ) with type [System.Object[]].
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 493
493:             , $result | Should -BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
```
FormatHex.Count and Offset parameters.Count = length
```powershell
Expected regular expression '00000000000000000000   4E 6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65  Now is the winte' to case sensitively match '0000000000000000   4E 6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65  Now is the winte', but it did not match.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 517
517:             $result[0].ToString() | Should -MatchExactly "00000000000000000000   4E 6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65  Now is the winte"
```
FormatHex.Count and Offset parameters.Count = 1
```powershell
Expected regular expression '00000000000000000000   4E                                               N               ' to case sensitively match '0000000000000000   4E                                               N               ', but it did not match.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 524
524:             $result.ToString() | Should -MatchExactly    "00000000000000000000   4E                                               N               "
```
FormatHex.Count and Offset parameters.Offset = length
```powershell
Expected an empty collection, but got collection with size 1 @().
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 533
533:             $result.Bytes | Should -HaveCount 0
```
FormatHex.Count and Offset parameters.Offset = 1
```powershell
Expected regular expression '00000000000000000001   6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65 72  ow is the winter' to case sensitively match '0000000000000001   6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65 72  ow is the winter', but it did not match.
at <ScriptBlock>, /home/vsts/work/1/s/test/powershell/Modules/Microsoft.PowerShell.Utility/Format-Hex.Tests.ps1: line 542
542:             $result[0].ToString() | Should -MatchExactly "00000000000000000001   6F 77 20 69 73 20 74 68 65 20 77 69 6E 74 65 72  ow is the winter"
```